src/codemirror/widget.ts

```diff
@@ -62,9 +76,34 @@ class CodeMirrorWidget extends Widget {
    }
 
   /**
+   * Handle the DOM events for the widget.
+   *
+   * @param event - The DOM event sent to the widget.
+   *
+   * #### Notes
+   * This method implements the DOM `EventListener` interface and is
+   * called in response to events on the notebook panel's node. It should
+   * not be called directly by user code.
+   */
+  handleEvent(event: Event): void {
+    switch (event.type) {
+    case 'focus':
+      this._evtFocus(event as MouseEvent);
+      break;
+    case 'blur':
+      this._evtBlur(event as MouseEvent);
+      break;
+    default:
+      break;
+    }
+  }
+
+  /**
    * A message handler invoked on an `'after-attach'` message.
    */
   protected onAfterAttach(msg: Message): void {
+    this.node.addEventListener('focus', this, true);
```

Comment:
What is the advantage of having `this` instead of calling `_evtFocus` directly? Why have an extra level of indirection?